Fix some missing code in step-by-step walk through (#10519)
PyTorchLightning/pytorch-lightning: docs/source/starter/introduction_guide.rst

            loss = F.nll_loss(logits, y)
            return loss

Optimizer
---------

> Next we choose what optimizer to use for training our system.

Reviewer: ```suggestion
Next, we choose which optimizer to use for training our model.
```

Reviewer: system isn't defined nor introduced.

In PyTorch we do it as follows:

.. code-block:: python

    from torch.optim import Adam

    optimizer = Adam(LitMNIST().parameters(), lr=1e-3)

> In Lightning we do the same but organize it under the :func:`~pytorch_lightning.core.LightningModule.configure_optimizers` method.

Reviewer: ```suggestion
In Lightning, this code is organized within the :func:`~pytorch_lightning.core.LightningModule.configure_optimizers` method of the LightningModule.
```

.. testcode::

    class LitMNIST(LightningModule):
        def configure_optimizers(self):
            return Adam(self.parameters(), lr=1e-3)

> .. note:: The LightningModule itself has the parameters, so pass in self.parameters()

Reviewer: ```suggestion
.. note:: The LightningModule is a subclassing :class:`~torch.nn.Module`  and therefore, you can access its children parameters with ``self.parameters()``.
```

> However, if you have multiple optimizers use the matching parameters

Reviewer: ```suggestion
If you have multiple optimizers, you can configure them as follows:
```

.. testcode::

    class LitMNIST(LightningModule):
        def configure_optimizers(self):
            return Adam(self.generator(), lr=1e-3), Adam(self.discriminator(), lr=1e-3)

Data
----

Install torchvision library.

.. code-block:: shell

>     pip install torchvision

Reviewer 2: Reviewers: Placing this installation guide here could be distracting to some users maybe? Instead, should there be an installation guide for all extra packages at the very top of the page?

e.g. 

```
Install extra packages used in this tutorial:

.. code-block:: shell

    pip install torchvision transformers ...

```

Reviewer 3: Agree this may be distracting and breaks the flow. Pretty sure this can be left out entirely. If the user does not know how to install this Python package while seeing an import error, they are probably not ready to use Lightning at this point anyway.

Reviewer 2: > Pretty sure this can be left out entirely.

That's what I thought, too. If we do this, we should do this to the entire docs for consistency, which can be quite redundant and less readable...

@os1ma Could you revert the change here at least for now?

Author: Thanks for the review!
I agree that this `pip install` explanation is not necessary for Lightning users. I will remove this part.

            loss = F.nll_loss(logits, y)
            return loss

> Optimizer

Reviewer 4: why is this moved? :rabbit:

Author: Hi @Borda. Thanks for the review!

If the original order that "Optimizer" section is after the "Data" section, the code in the "Data" section fails to execute. This is because the required configure_optimizers method is defined in the "Optimizer" section after the "Data" section.

So it may be good that we swap the order with "Data" section and "Optimizer" section to avoid the problem.

If it is not reasonable, I'll be happy to fix the change!

.. code-block:: python

    from pytorch_lightning import Trainer

> 

Reviewer 2: Reviewers: Instead of having import statements scattered in the page, should we have a single place to import all functions/classes at the very top of this page maybe? Similarly to https://pytorch-lightning.readthedocs.io/en/latest/starter/new-project.html#step-0-install-pytorch-lightning

Reviewer 3: Agree. The imports should be done once in the first example where they fit in and then not be repeated. If a new section appears that needs new imports, we add them.

The individual code snippets are anyway not runnable standalone and have to be taken with the context of the whole page.

Author: Thanks for this comment too!
I will move the import to the other imports location at the top of the section.

            loss = F.nll_loss(logits, y)
            return loss

        def configure_optimizers(self):
            return Adam(self.parameters(), lr=1e-3)

> 

Reviewer 2: Great fix :)
